### ISSUE TITLE
Lag causes potions being applied twice and hatching twice

### ISSUE THREAD

**Issue by LinusS1** [hackagotchi/hackagotchi#75]
Due to the app page not being updated, I was able to apply a potion twice. 
<img width="702" alt="Screen Shot 2020-06-27 at 11 47 23 AM" src="https://user-images.githubusercontent.com/19542092/85930023-e1f38380-b86d-11ea-92c0-e175f3f4e671.png">
It seems like the boosts are correct for just one potion, so no harm done (though this could be another bug where you can only have one potion applied at a time).

I also stumbled across the lag in updating bug yesterday where (I believe) I was able to hatch a bagling egg twice. I had the modal open, and after the first hatch, I clicked hatch again and I was able to hatch it.

Solutions: (I don't know if these have been talked about): update the app home as soon as the potion/item is used. Or just validate that it hasn't been used.

**Comment by Muirrum** [comment]
I think this is a result of the time between game ticks that was brought up a few days ago. if we resolve that issue then we can queue tasks faster to be sent out to update the game window.

**Comment by pradyungn** [comment]
Pretty much what muirrum said - @Muirrum maybe verify the tasks in the queue before executing them. I dont' think that the game ticks themselves can be fixed, but this is something we've gone over beofre, the commands coming in from slack need to be validated at some point.

**Comment by Muirrum** [comment]
I think if we add a small "verify" method to run before any task is
executed that would work.

**Comment by pradyungn** [comment]
yeah that should work

**Comment by Muirrum** [comment]
Yeah, and I think is something that's going to wait for the backend implementation to be done.

**Comment by lukec11** [comment]
Similar to #117  - This needs validation on the backend. It could also be caused by lag on the slack side - if they are delayed in sending requests to us to update the app home page.

**Comment by nzeng8** [comment]
Since I've gotten 20 potions, there have been three appearing instead of two for each potion applied...

**Comment by lukec11** [comment]
> Since I've gotten 20 potions, there have been three appearing instead of two for each potion applied...

Can you explain what you mean @nzeng8 ? If you're saying that multiple potions show up when you apply one, that's a separate issue. This issue surrounds game tick time + verification of requests to ensure no duplication.

**Comment by nzeng8** [comment]
The bug that I'm facing is that when one potion is applied, 3 potions show up next to the plant as if 3 are applied. However, according to the Yield Stats, it only shows the effect of one potion. This is a visual issue.
